src/lib/Save.js

```diff
@@ -0,0 +1,32 @@
+export default class Save {
+  constructor(logs) {
+    this._logs = logs;
+  }
+
+  downloadFile(blob, filename) {
+    const url = URL.createObjectURL(blob);
+
+    function listener(result) {
+      if (result.state.current === 'complete') {
+        setTimeout(() => {
+          URL.revokeObjectURL(url);
+        }, 100);
+        browser.downloads.onChanged.removeListener(listener);
+      }
+    }
+
+    browser.downloads.onChanged.addListener(listener);
+
+    browser.downloads.download({
+      url,
+      filename,
+    });
```

Comment:
[`downloads.download` calls should resolve to a download id](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download#Return_value), I think that we may want to use that download id to be sure that the onChanged events are actually related to the download triggered by this downloads.download API call.

Also, based on the docs, there is a chance that downloads.download may reject because the download has been immediately interrupted, and in that case we may want to do some cleanup (e.g. remove the onChanged listener and clear the blob url) and report the error to the caller.

are there other error scenarios that we may need to take into account?

Reply:
> [`downloads.download` calls should resolve to a download id](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download#Return_value), I think that we may want to use that download id to be sure that the onChanged events are actually related to the download triggered by this downloads.download API call.
Yeah, I get a id when the promise is resolved. I will make this chance :-)

> Also, based on the docs, there is a chance that downloads.download may reject because the download has been immediately interrupted, and in that case we may want to do some cleanup (e.g. remove the onChanged listener and clear the blob url) and report the error to the caller.
> 
> are there other error scenarios that we may need to take into account?
Thanks for noting this, going to cover this above corner case in next iteration and I will checkout if there is any other scenarios, we should take care of.